src/mrinufft/trajectories/display3D.py

```diff
@@ -0,0 +1,140 @@
+)
+import numpy as np
+from typing import Tuple
+from mrinufft.io import read_trajectory
+
+
+def get_gridded_trajectory(
+    shots: np.ndarray,
+    shape: Tuple,
+    grid_type: str = "density",
+    osf: int = 1,
+    backend: str = "gpunufft",
+    traj_params: dict = None,
+    turbo_factor: int = 176,
+    elliptical_samp: bool = True,
+    threshold: float = 1e-3,
+):
+    """
+    Compute the gridded trajectory for MRI reconstruction.
+
+    This function helps in gridding a k-space sampling trajectory to a desired shape.
+    The gridding process can be carried out to reflect the sampling density,
+    sampling time, inversion time, k-space holes, gradient strengths, or slew rates.
+    Please check `grid_type` parameter to know the benefits of each type of gridding.
+
+    Parameters
+    ----------
+    shots : ndarray
+        The input array of shape (N, M, D), where N is the number of shots and M is the
+        number of samples per shot and D is the dimension of the trajectory (usually 3)
+    shape : tuple
+        The desired shape of the gridded trajectory.
+    grid_type : str, optional
+        The type of gridded trajectory to compute. Default is "density".
+        It can be one of the following:
+        "density" : Get the sampling density in closest number of samples per voxel.
+            Helps understand suboptimal sampling, by showcasing regions with strong
+            oversampling.
+        "time" : Showcases when the k-space data is acquired in time.
+            This is helpful to view and understand off-resonance effects.
+            Generally, lower off-resonance effects occur when the sampling trajectory
+            has smoother k-space sampling time over the k-space.
+        "inversion" : Relative inversion time at the sampling location. Needs
+            turbo_factor to be set. This is useful for analyzing the exact inversion
+            time when the k-space is acquired, for sequences like MP(2)RAGE.
+        "holes": Show the k-space holes within a ellipsoid of the k-space.
+        "gradients": Show the gradient strengths of the k-space trajectory.
+        "slew": Show the slew rate of the k-space trajectory.
+    osf : int, optional
+        The oversampling factor for the gridded trajectory. Default is 1.
+    backend : str, optional
+        The backend to use for gridding. Default is "gpunufft".
+        Note that "gpunufft" is anyway used to get the `pipe` density internally.
+    traj_params : dict, optional
+        The trajectory parameters. Default is None.
+        This is only needed when `grid_type` is "gradients" or "slew".
+        The parameters needed include `img_size`, `FOV`, and `gamma` of the sequence.
+        Generally these values are stored in the header of the trajectory file.
+    turbo_factor : int, optional
+        The turbo factor when sampling is with inversion. Default is 176, which is
+        the default turbo factor for MPRAGE acquisitions at 1mm whole
```

Comment:
Please rephrase into a complete sentence. Also it tells when it is relevant but we still don't know what it is